basic-javascript/project-1.spec.js

```diff
@@ -1,3 +1,255 @@
+            expect(helpers.square(5)).toEqual(25);
+        })
+        it('should throw err when calling other than number', () => {
+            expect(() => {
+                helpers.square('ten')
+            }).toThrow();
+        })
+    })
+
+    describe('cube() ', () => {
+        it('should be able to correctly square', () => {
+            expect(helpers.cube(5)).toEqual(125);
+        })
+        it('should throw err when calling other than number', () => {
+            expect(() => {
+                helpers.cube('ten')
+            }).toThrow();
+        })
+    })
+
+    describe('raiseToPower() ', () => {
+        it('should be able to correctly raiseToPower', () => {
+            expect(helpers.raiseToPower(5, 2)).toEqual(25);
+        })
+        it('should throw err when calling other than number', () => {
+            expect(() => {
+                helpers.raiseToPower('ten', 2)
+            }).toThrow();
+        })
+    })
+
+    describe('roundNumber() ', () => {
+        it('should be able to correctly round number', () => {
+            expect(helpers.roundNumber(5.25)).toEqual(5);
+        })
+        it('should throw err when calling other than number', () => {
+            expect(() => {
+                helpers.roundNumber('ten')
+            }).toThrow();
+        })
+    })
+
+    describe('roundUp() ', () => {
+        it('should be able to correctly roundUp', () => {
+            expect(helpers.roundUp(5.25)).toEqual(6);
+        })
+        it('should throw err when calling other than number', () => {
+            expect(() => {
+                helpers.roundUp('ten')
+            }).toThrow();
+        })
+    })
+
+    describe('addExclamationPoint() ', () => {
+        it('should be able to correctly addExclamationPoint', () => {
+            expect(helpers.addExclamationPoint('hello')).toEqual('hello!');
+        })
+    })
+
+    describe('combineNames() ', () => {
+        it('should be able to correctly combine Name', () => {
+            expect(helpers.combineNames('yanrong', 'xu')).toEqual('yanrong xu');
+        })
+    })
+
+    describe('getGreeting() ', () => {
+        it('should be able to correctly getGreeting', () => {
+            expect(helpers.getGreeting('yanrong')).toEqual('Hello yanrong!');
+        })
+    })
+
+    describe('getRectangleArea() ', () => {
+        it('should be able to correctly get rectangle Area', () => {
+            expect(helpers.getRectangleArea(4,5)).toEqual(20);
+        })
+        it('should throw err when calling other than number', () => {
+            expect(() => {
+                helpers.getRectangleArea('ten', 5)
+            }).toThrow();
+        })
+    })
+
+    describe('getTriangleArea() ', () => {
+        it('should be able to correctly get triangle Area', () => {
+            expect(helpers.getTriangleArea(4,5)).toEqual(10);
+        })
+        it('should throw err when calling other than number', () => {
```

Comment:
Could check how it handles a negative number for geometrical equations, and make sure you like how it behaves